libs/dh/imbalance-prices/shell/src/lib/drawer/dh-drawer.component.ts

```diff
@@ -109,50 +109,48 @@ import { format } from 'date-fns';
     WattButtonComponent,
     WattSpinnerComponent,
     VaterFlexComponent,
+
     DhStatusBadgeComponent,
     DhEmDashFallbackPipe,
     DhTableDayViewComponent,
-    DhFeatureFlagDirective,
   ],
 })
 export class DhImbalancePricesDrawerComponent {
   private readonly toastService = inject(WattToastService);
   private readonly httpClient = inject(ImbalancePricesHttp);
   private readonly apollo = inject(Apollo);
 
-  @ViewChild(WattDrawerComponent)
-  drawer: WattDrawerComponent | undefined;
+  private readonly year = computed(() => Number(format(this.imbalancePrice()!.name, 'yyyy')));
+  private readonly month = computed(() => Number(format(this.imbalancePrice()!.name, 'M')));
 
   imbalancePrice = input<DhImbalancePrice>();
+
   imbalancePricesForMonth = signal<DhImbalancePricesForMonth[]>([]);
   isLoading = signal(false);
+  lastUpdated = computed(() => {
+    const [firstDay] = this.imbalancePricesForMonth();
 
-  @Output() closed = new EventEmitter<void>();
-
-  query = this.apollo.watchQuery({
-    useInitialLoading: true,
-    notifyOnNetworkStatusChange: true,
-    query: GetImbalancePricesMonthOverviewDocument,
-    variables: undefined,
+    return firstDay?.importedAt;
   });
 
+  @ViewChild(WattDrawerComponent)
+  drawer: WattDrawerComponent | undefined;
+
+  @Output() closed = new EventEmitter<void>();
+
   constructor() {
-    effect((onCleanup) => {
-      if (this.imbalancePrice()) {
-        this.drawer?.open();
-
-        this.query.setVariables({
-          year: this.imbalancePrice()!.name.getFullYear(),
-          // Add 1 because months in the backend start at 1
-          month: this.imbalancePrice()!.name.getMonth() + 1,
-          areaCode: this.imbalancePrice()!.priceAreaCode,
-        });
+    effect(
+      (onCleanup) => {
+        if (this.imbalancePrice()) {
+          this.drawer?.open();
 
-        const subscription = this.fetchData();
+          const subscription = this.fetchData();
 
-        onCleanup(() => subscription.unsubscribe());
-      }
-    });
+          onCleanup(() => subscription.unsubscribe());
```

Comment:
Suggestion: Dont think you need a unsubscribe here, because you use apollo query and not queryWatch

Reply:
It's a bit weird. The request is not unsubscribed if the drawer is closed before the request completes. But then the next time the drawer is opened, the request completes immediately. So it doesn't look the logic above is necessary because it doesn't make any difference.